WFD/PreviewItem.py

```diff
@@ -0,0 +1,256 @@
+                    prop=prop,
+                    value=PreviewItem.format_itis(statement),
+                    quals=' \n'.join(quals),
+                    references=''  # self.ref  # re-add if multiple references
+                )
+
+        table += table_end
+        return table
+
+    @staticmethod
+    def format_qual(qual):
+        """Create a preview for a single Qualifier."""
+        prop = PreviewItem.make_wikidata_template(qual.prop)
+        itis = PreviewItem.format_itis(qual.itis)
+        return '{0}: {1}'.format(prop, itis)
+
+    @staticmethod
+    def format_reference(ref):
+        """Create a preview for a single Reference."""
+        txt = ''
+        if ref.source_test:
+            txt += ':{}:\n'.format(PreviewItem.make_text_italics('tested'))
+            for claim in ref.source_test:
+                txt += ':*{}\n'.format(PreviewItem.format_claim(claim))
+        if ref.source_notest:
+            txt += ':{}:\n'.format(PreviewItem.make_text_italics('not tested'))
+            for claim in ref.source_notest:
+                txt += ':*{}\n'.format(PreviewItem.format_claim(claim))
+
+        return txt
+
+    @staticmethod
+    def format_claim(claim):
+        """Create a preview for a single pywikibot.Claim."""
+        special = False
+        itis = claim.getTarget()
+        if claim.getSnakType() != 'value':
+            special = True
+            itis = claim.getSnakType()
+
+        return '{prop}: {itis}'.format(
+            prop=PreviewItem.make_wikidata_template(claim.id),
+            itis=PreviewItem.format_itis(itis, special)
+        )
+
+    @staticmethod
+    def format_itis(itis, special=False):
+        """
+        Create a preview for the itis component of a Statement.
+
+        :param itis: a Statement or the itis component of a Statement or
+            Qualifier.
+        :param special: if it is a special type of value (i.e. novalue,
+            somevalue). Is set automatically if a Statement is passed to itis.
+        """
+        # handle the case where a Statement was passed
+        if isinstance(itis, WdS.Statement):
+            special = itis.special
+            itis = itis.itis
+
+        if isinstance(itis, pywikibot.ItemPage):
+            return PreviewItem.make_wikidata_template(itis)
+        elif special:
+            return PreviewItem.make_wikidata_template(itis, special=True)
+        elif isinstance(itis, pywikibot.WbQuantity):
+            unit = itis.get_unit_item() or ''
+            amount = itis.amount
+            if unit:
+                unit = PreviewItem.make_wikidata_template(unit)
+            return '{} {}'.format(amount, unit).strip()
+        elif isinstance(itis, pywikibot.WbTime):
+            return itis.toTimestr()
+        else:
+            return str(itis)
+
+    def format_labels(self):
+        """
+        Create a label(s) preview.
+
+        The first label is italicised (to indicate it is the preferred label
```

Comment:
I broke this out as [T167791](https://phabricator.wikimedia.org/T167791). Separating per line is probably the best solution but I want to ensure it works (without swamping the preview page) even when there are multiple languages.